aten/src/ATen/native/Pow.cpp

```diff
@@ -6,85 +6,65 @@
 #include <ATen/ScalarOps.h>
 #include <ATen/native/Resize.h>
 
-namespace at { namespace native {
+namespace at {
+namespace meta {
 
-DEFINE_DISPATCH(pow_tensor_tensor_stub);
-DEFINE_DISPATCH(pow_tensor_scalar_stub);
-
-Tensor& pow_out(Tensor& result, const Tensor& base, const Tensor& exp) {
-  if (exp.dim() == 0 && exp.device().is_cpu() && base.is_cuda()) {
-    return native::pow_out(result, base, exp.item());
-  }
-  auto iter = TensorIterator::binary_op(result, base, exp);
-  pow_tensor_tensor_stub(iter.device_type(), iter);
-  return result;
+TORCH_META_FUNC2(pow, Tensor_Tensor) (const Tensor& base, const Tensor& exp) {
+  build_binary_op(maybe_get_output(), base, exp);
 }
 
-Tensor& pow_out(Tensor& result, const Tensor& base, Scalar exp) {
+TORCH_META_FUNC2(pow, Tensor_Scalar) (const Tensor& base, const Scalar exp) {
   // Numpy compatibility check:
   TORCH_CHECK(!(isIntegralType(base.scalar_type(), true) &&
               exp.isIntegral(true) && exp.toLong() < 0),
               "Integers to negative integer powers are not allowed.");
 
   auto common_dtype = at::result_type(base, exp);
-  TORCH_CHECK(at::can_cast(common_dtype, result.scalar_type()),
-           "result type ", common_dtype, " can't be cast to the desired output type ",
-           result.scalar_type());
+  build_unary_op(maybe_get_output(), base.to(common_dtype));
+}
+
+TORCH_META_FUNC2(pow, Scalar) (const Scalar base, const Tensor& exp) {
+    // This overload doesn't directly use TensorIterator. It attempts to short-circuit,
+    // but otherwise redispatches to the Tensor_Tensor overload.
+    auto dtype = at::result_type(base, exp);
+    // Need to call the set_output() overload of set_output to forward named tensor information.
```

Comment:
Well, of course you need to call `set_output`, how else is the operator gonna know what the output size is! (Difference with functions above is `build_unary_op` implicitly calls `set_output` in its guts)

Reply:
oh yeah, my intention with this comment was calling out that we needed to use the "internal guts" overload of `set_output()` since we want to pass names through. Probably not a very useful comment though